Cannot find module '@directus/types' or its corresponding type declarations - Using angular 16 - Ionic 7 as Frontend

### Describe the Bug

I am using Angular 16 - ionic 7 as frontend.
while using the latest directus sdk 11.0.0 i got the below error

Error: node_modules/@directus/sdk/dist/output-476d05a4.d.ts:1:28 - error TS2307: Cannot find module '@directus/types' or its corresponding type declarations.

1 import { PrimaryKey } from '@directus/types';

![Screenshot 2023-07-27 110358](https://github.com/directus/directus/assets/10194474/705c3b7b-228a-4ada-859e-f185690ac4dd)

Then i installed npm i @directus/types
New error as 
![Screenshot 2023-07-27 110420](https://github.com/directus/directus/assets/10194474/52259f15-229f-4b53-8bca-43830bf6ca95)


### To Reproduce

Fresh ionic app creation : 
Ionic start --type=angular tabs

Install Directus sdk : 
npm install @directus/sdk

Start the app - after importing sdk in appropriate service / component
ionic start
![Screenshot 2023-07-27 111043](https://github.com/directus/directus/assets/10194474/d065aadf-e2ee-4756-9415-358ba8807260)


### Directus Version

10.5.0

### Hosting Strategy

Self-Hosted (Custom)

--- a/.changeset/itchy-impalas-dream.md
+++ b/.changeset/itchy-impalas-dream.md
@@ -1,0 +1,5 @@
+---
+"@directus/sdk": patch
+---
+
+Minor SDK fixes

--- a/sdk/package.json
+++ b/sdk/package.json
@@ -48,12 +48,14 @@
 	},
 	"devDependencies": {
 		"@directus/tsconfig": "workspace:*",
-		"@directus/types": "workspace:*",
 		"@types/node-fetch": "2.6.4",
 		"@vitest/coverage-c8": "0.31.1",
 		"tsup": "7.1.0",
 		"typescript": "5.0.4",
 		"vitest": "0.31.1"
+	},
+	"dependencies": {
+		"@directus/types": "workspace:*"
 	},
 	"engines": {
 		"node": ">=18.0.0"

--- a/sdk/src/auth/composable.ts
+++ b/sdk/src/auth/composable.ts
@@ -123,6 +123,9 @@ export const authentication = (mode: AuthenticationMode = 'cookie', config: Auth
 
 				const data = await request<AuthenticationData>(requestUrl.toString(), {
 					method: 'POST',
+					headers: {
+						'Content-Type': 'application/json',
+					},
 					body: JSON.stringify({
 						email,
 						password,
@@ -137,6 +140,9 @@ export const authentication = (mode: AuthenticationMode = 'cookie', config: Auth
 
 				const options: RequestInit = {
 					method: 'POST',
+					headers: {
+						'Content-Type': 'application/json',
+					},
 				};
 
 				if (mode === 'json' && authData?.refresh_token) {

--- a/sdk/src/auth/composable.ts
+++ b/sdk/src/auth/composable.ts
@@ -1,3 +1,4 @@
+import type { loginOptions } from '../index.js';
 import type { DirectusClient } from '../types/client.js';
 import { getRequestUrl } from '../utils/get-request-url.js';
 import { request } from '../utils/request.js';
@@ -115,21 +116,23 @@ export const authentication = (mode: AuthenticationMode = 'cookie', config: Auth
 
 		return {
 			refresh,
-			async login(email: string, password: string) {
+			async login(email: string, password: string, options: loginOptions = {}) {
 				// TODO: allow for websocket only authentication
 				resetStorage();
 
-				const requestUrl = getRequestUrl(client.url, '/auth/login');
+				const authPath = options.provider ? `/auth/login/${options.provider}` : '/auth/login';
+				const requestUrl = getRequestUrl(client.url, authPath);
+
+				const authData: Record<string, string> = { email, password };
+				if ('otp' in options) authData['otp'] = options.otp;
+				if ('mode' in options) authData['mode'] = options.mode;
 
 				const data = await request<AuthenticationData>(requestUrl.toString(), {
 					method: 'POST',
 					headers: {
 						'Content-Type': 'application/json',
 					},
-					body: JSON.stringify({
-						email,
-						password,
-					}),
+					body: JSON.stringify(authData),
 				});
 
 				setCredentials(data);

--- a/sdk/src/auth/types.ts
+++ b/sdk/src/auth/types.ts
@@ -1,3 +1,5 @@
+import type { loginOptions } from "../index.js";
+
 export type AuthenticationMode = 'json' | 'cookie';
 
 export interface AuthenticationData {
@@ -19,7 +21,7 @@ export interface AuthenticationConfig {
 }
 
 export interface AuthenticationClient<_Schema extends object> {
-	login(email: string, password: string): Promise<AuthenticationData>;
+	login(email: string, password: string, options: loginOptions): Promise<AuthenticationData>;
 	refresh(): Promise<AuthenticationData>;
 	logout(): Promise<void>;
 

--- a/sdk/src/auth/types.ts
+++ b/sdk/src/auth/types.ts
@@ -1,4 +1,4 @@
-import type { loginOptions } from "../index.js";
+import type { loginOptions } from '../index.js';
 
 export type AuthenticationMode = 'json' | 'cookie';
 

--- a/sdk/src/rest/commands/auth/login.ts
+++ b/sdk/src/rest/commands/auth/login.ts
@@ -20,7 +20,7 @@ export const login =
 	<Schema extends object>(
 		email: string,
 		password: string,
-		options: loginOptions
+		options: loginOptions = {}
 	): RestCommand<AuthenticationData, Schema> =>
 	() => {
 		const path = options.provider ? `/auth/login/${options.provider}` : '/auth/login';

--- a/sdk/src/rest/commands/create/collections.ts
+++ b/sdk/src/rest/commands/create/collections.ts
@@ -1,5 +1,5 @@
 import type { DirectusCollection } from '../../../schema/collection.js';
-import type { ApplyQueryFields, Query } from '../../../types/index.js';
+import type { ApplyQueryFields, NestedPartial, Query } from '../../../types/index.js';
 import type { RestCommand } from '../../types.js';
 
 export type CreateCollectionOutput<
@@ -18,7 +18,7 @@ export type CreateCollectionOutput<
  */
 export const createCollection =
 	<Schema extends object, const TQuery extends Query<Schema, DirectusCollection<Schema>>>(
-		item: Partial<DirectusCollection<Schema>>,
+		item: NestedPartial<DirectusCollection<Schema>>,
 		query?: TQuery
 	): RestCommand<CreateCollectionOutput<Schema, TQuery>, Schema> =>
 	() => ({

--- a/sdk/src/rest/commands/create/relations.ts
+++ b/sdk/src/rest/commands/create/relations.ts
@@ -16,7 +16,9 @@ export type CreateRelationOutput<
  * @returns Returns the relation object for the created relation.
  */
 export const createRelation =
-	<Schema extends object>(item: NestedPartial<DirectusRelation<Schema>>): RestCommand<CreateRelationOutput<Schema>, Schema> =>
+	<Schema extends object>(
+		item: NestedPartial<DirectusRelation<Schema>>
+	): RestCommand<CreateRelationOutput<Schema>, Schema> =>
 	() => ({
 		path: `/relations`,
 		body: JSON.stringify(item),

--- a/sdk/src/rest/commands/create/relations.ts
+++ b/sdk/src/rest/commands/create/relations.ts
@@ -1,5 +1,5 @@
 import type { DirectusRelation } from '../../../schema/relation.js';
-import type { ApplyQueryFields } from '../../../types/index.js';
+import type { ApplyQueryFields, NestedPartial } from '../../../types/index.js';
 import type { RestCommand } from '../../types.js';
 
 export type CreateRelationOutput<
@@ -16,7 +16,7 @@ export type CreateRelationOutput<
  * @returns Returns the relation object for the created relation.
  */
 export const createRelation =
-	<Schema extends object>(item: Partial<DirectusRelation<Schema>>): RestCommand<CreateRelationOutput<Schema>, Schema> =>
+	<Schema extends object>(item: NestedPartial<DirectusRelation<Schema>>): RestCommand<CreateRelationOutput<Schema>, Schema> =>
 	() => ({
 		path: `/relations`,
 		body: JSON.stringify(item),

--- a/sdk/src/rest/commands/update/collections.ts
+++ b/sdk/src/rest/commands/update/collections.ts
@@ -1,5 +1,5 @@
 import type { DirectusCollection } from '../../../schema/collection.js';
-import type { ApplyQueryFields, Query } from '../../../types/index.js';
+import type { ApplyQueryFields, NestedPartial, Query } from '../../../types/index.js';
 import type { RestCommand } from '../../types.js';
 
 export type UpdateCollectionOutput<
@@ -18,7 +18,7 @@ export type UpdateCollectionOutput<
 export const updateCollection =
 	<Schema extends object, const TQuery extends Query<Schema, DirectusCollection<Schema>>>(
 		collection: DirectusCollection<Schema>['collection'],
-		item: Partial<DirectusCollection<Schema>>,
+		item: NestedPartial<DirectusCollection<Schema>>,
 		query?: TQuery
 	): RestCommand<UpdateCollectionOutput<Schema, TQuery>, Schema> =>
 	() => ({

--- a/sdk/src/rest/commands/update/relations.ts
+++ b/sdk/src/rest/commands/update/relations.ts
@@ -1,5 +1,5 @@
 import type { DirectusRelation } from '../../../schema/relation.js';
-import type { ApplyQueryFields, Query } from '../../../types/index.js';
+import type { ApplyQueryFields, NestedPartial, Query } from '../../../types/index.js';
 import type { RestCommand } from '../../types.js';
 
 export type UpdateRelationOutput<
@@ -20,7 +20,7 @@ export const updateRelation =
 	<Schema extends object, const TQuery extends Query<Schema, DirectusRelation<Schema>>>(
 		collection: DirectusRelation<Schema>['collection'],
 		field: DirectusRelation<Schema>['field'],
-		item: Partial<DirectusRelation<Schema>>,
+		item: NestedPartial<DirectusRelation<Schema>>,
 		query?: TQuery
 	): RestCommand<UpdateRelationOutput<Schema, TQuery>, Schema> =>
 	() => ({

--- a/sdk/src/rest/composable.ts
+++ b/sdk/src/rest/composable.ts
@@ -17,9 +17,13 @@ export const rest = (config: RestConfig = {}) => {
 				const options = getOptions();
 
 				// all api requests require this content type
-				if (!options.headers?.['Content-Type']) {
-					if (!options.headers) options.headers = {};
+				if (!options.headers) options.headers = {};
+
+				if ('Content-Type' in options.headers === false) {
 					options.headers['Content-Type'] = 'application/json';
+				} else if (options.headers['Content-Type'] === 'multipart/form-data') {
+					// let the fetch function deal with multipart boundaries
+					delete options.headers['Content-Type'];
 				}
 
 				// we need to use THIS here instead of client to access overridden functions

--- a/sdk/src/utils/request.ts
+++ b/sdk/src/utils/request.ts
@@ -26,11 +26,6 @@ export const request = async <Output = any>(
 
 	const outputFormatter = formatter !== undefined && formatter !== null ? formatter : defaultFormatter;
 
-	// use json content by default but allow overrides
-	if ('Content-Type' in headers === false) {
-		headers['Content-Type'] = 'application/json';
-	}
-
 	options.headers = headers;
 
 	const response = await globalThis

--- a/sdk/tsup.config.js
+++ b/sdk/tsup.config.js
@@ -9,7 +9,6 @@ export default defineConfig(() => ({
 	format: ['cjs', 'esm'], // generate cjs and esm files
 	minify: env === 'production',
 	bundle: env === 'production',
-	skipNodeModulesBundle: true,
 	watch: env === 'development',
 	target: 'es2020',
 	entry: [

